projects/client/RabbitMQ.Client/src/client/impl/TcpClientAdapter.cs

```diff
@@ -27,10 +27,17 @@ public TcpClientAdapter(Socket socket)
             AssertSocket();
             var adds = await Dns.GetHostAddressesAsync(host).ConfigureAwait(false);
             var ep = TcpClientAdapterHelper.GetMatchingHost(adds, sock.AddressFamily);
-            if (ep == default(IPAddress))
+
+            if (ep == null)
```

Comment:
There is no need to change this from `if (ep == default(IPAddress))`

Reply:
I would say it is incorrect to check for null with default approach here.

Reply:
Incorrect how?

Reply:
IPAddress is a reference type. So, in my point of view checking on null with reference type should be with "== null". It is clear to read and it should be a very good reason to use "== default". But anyway, I am not a contributor and I am not going to start explaining what the best practices are.